docs/translations.md

```diff
@@ -0,0 +1,30 @@
+# Managing translations
+
+Right now we're using the `resourceConfigurations` mechanism provided by the Android Gradle Plugin to limit which
```

Comment:
This is used to generally exclude all kinds of resources from the actual build. As I understand we use it to filter out languages that are not translated well enough. This might be necessary as Weblate just pushes every language regardeless of it's translation state. Additionally we manage `supported_languages` to reflect this to the available in app languages. I think this whole could be improved.

I originally just wanted to delete the unused translations as Weblate doesn't offer a way to define a criteria for pushing translations. As a result everything added to weblate is pushed to the project. That includes even translations that fail the weblate checks. I personally used different translations tools that offered a set of checks and criterias prior pushing, ensuring a clean project state. But as we have choosen weblate, we need to accomodate for this.

We could alter the approach and restrict `resourceConfigurations` to release builds only. This offers the posibility to keep good translations for production and allow partial translations to be visible in a debug build. Hence allowing translators to see their translations in real world before reaching a certain translation percentage.

This leaves the `supported_languages` still unaddressed. Here we could follow Android 13's [per-app lanugage preferences](https://developer.android.com/guide/topics/resources/app-languages) instead of our custom solution. It is backwards compatible using the app compat library. Little downside is it's Activty context requirement.

Reply:
> This is used to generally exclude all kinds of resources from the actual build. As I understand we use it to filter out languages that are not translated well enough. This might be necessary as Weblate just pushes every language regardeless of it's translation state. 

We'd still want to use this mechanism because libraries might come with translations for languages we don't support. Without this, users could end up with parts of the UI being in their system language (that is not supported by the app) and the rest in our default language (English).

> We could alter the approach and restrict resourceConfigurations to release builds only. This offers the posibility to keep good translations for production and allow partial translations to be visible in a debug build. Hence allowing translators to see their translations in real world before reaching a certain translation percentage.

I'd go further and include all translations in betas and nightly builds (once we have them). However, it doesn't feel like this should be a priority right now.

Reply:
I know that it's also used to guard against translations introduced by 3rd-party libaries. But this is usually a sign of bad library design. Especialy in case there is a mismatch of supported languages between app and library, it's a hurdle to fill the gaps.

I had a quick look, which of our dependencies introduce such resources:
- `SearchPrefences` comes with `en, de` -> save
- Takisoft's `ColorPicker` and `DateTimePicker` come with a couple of translations -> we should replace them with another library -> should be replaced

Reply:
> I'd go further and include all translations in betas and nightly builds (once we have them). However, it doesn't feel like this should be a priority right now.

Yeah, `nightly` (daily) should have them for sure. `beta` at the monent not, we first need to have a separated beta build.